modules/loggers/mod_log_config.c

```diff
@@ -1666,13 +1781,279 @@ static ap_log_writer *ap_log_set_writer(ap_log_writer *handle)
     return old;
 }
 
+#if APU_MAJOR_VERSION > 1 || (APU_MAJOR_VERSION == 1 && APU_MINOR_VERSION >= 7)
+static ap_log_formatted_data * ap_json_log_formatter( request_rec *r,
+                           void *handle,
+                           ap_log_formatted_data *lfd,
+                           const void *itms)
+
+{
+    json_log_formatter_options * formatter_options = (json_log_formatter_options *) handle;
+    log_format_item *items = (log_format_item *) itms;
```

Comment:
I don't think an `ap_log_formatter` function should modify its `handle` parameter so it could be `const void *handle`?
Maybe the `lfd` could be returned as-is and not copied by some implementation (not here) so it makes sense to leave it non-const, or return a `const ap_log_formatted_data*` dunno.
Anyway, I think these locals should be const here:
```
    const json_log_formatter_options *formatter_options = handle;
    const log_format_item *items = itms;
```

Reply:
fixed